fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainerServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package org.fcrepo.kernel.impl;
+
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import static org.fcrepo.persistence.api.Options.Option.ARCHIVAL_GROUP;
+
+import org.fcrepo.kernel.api.FedoraSession;
+import org.fcrepo.kernel.api.models.Container;
+import org.fcrepo.kernel.api.services.ContainerService;
+import org.fcrepo.persistence.api.Options;
+import org.fcrepo.persistence.api.PersistentResource;
+import org.fcrepo.persistence.api.PersistentStorage;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+
+/**
+ * @author dbernstein
+ */
+public class ContainerServiceImpl implements ContainerService {
```

Comment:
One thing to think about is breaking free of the rdf == container paradigm that Fedora currently has.  When modeling in OCFL, for example, a container is analogous to a directory (with some associated RDF).  This isn't always desirable.  Sometimes you'd just want rdf as rdf/file; in LDP an RDFSource.